File: LeetCode/0053. Maximum Subarray.cpp
class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        int maxSum = nums[0], curSum = nums[0];
        for (int i = 1; i < nums.size(); i++)
        {
            if (curSum < 0)
                curSum = 0;
            curSum += nums[i];
            maxSum = max(curSum, maxSum);
        }
        return maxSum;
    }
};

File: LeetCode/0053. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        cur_sum = nums[0]
        for i in range(1, len(nums)):
            if cur_sum < 0:
                cur_sum = 0
            cur_sum += nums[i]
            max_sum = max(cur_sum, max_sum)
        return max_sum
